jupyter_publishing_service/app.py

```diff
@@ -88,18 +93,25 @@ def _validate_ip(self, proposal):
             value = ""
         return value
 
+    authenticator: AuthorizerABC = Instance(
+        klass="jupyter_publishing_service.authenticator.abc.AuthenticatorABC", allow_none=True
+    )
+
+    storage_manager: StorageManagerABC = Instance(
+        klass="jupyter_publishing_service.storage.abc.StorageManagerABC", allow_none=True
+    )
+
     def init_configurables(self):
-        self.file_manager = self.file_manager_class(parent=self, log=self.log)
-        self.collaborator_store = self.collaborator_store_class(parent=self, log=self.log)
         self.authenticator = self.authenticator_class(parent=self, log=self.log)
-        self.authorizer = self.authorizer_class(parent=self, log=self.log)
         set_authenticator_class(self.authenticator)
```

Comment:
Why not use the same pattern for authenticator as with other classes? i.e using the @default decorator

Reply:
You're totally right. We should use the `@default` pattern here as well. I'll update.